File: src/main/java/com/finfamily/domain/Groups.java
package com.finfamily.domain;

import io.swagger.annotations.ApiModelProperty;

import javax.persistence.*;
import java.util.Set;

@Entity
@Embeddable
@Table(name = "groups")
public class Groups {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ApiModelProperty(hidden = true , required = false)
    private int id;

    @Column(name = "group_name")
    private String groupName;

//    @OneToMany(mappedBy = "groups", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
//    private Set<GroupParticipants> groupParticipants;

    @OneToOne(mappedBy = "groups", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private GroupWallet groupWallet;

//    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
//    @JoinColumn(name = "group_type")
//    private GroupTypes groupTypes;

    @Column(name = "group_type")
    private int groupType;

    @Column(name = "group_owner")
    private int groupOwner;

    public Groups (int id, String groupName, int groupType, int groupOwner){
        this.id = id;
        this.groupName = groupName;
        this.groupType = groupType;
        this.groupOwner = groupOwner;
    }

    public int getId() {
        return id;
    }

    public String getGroupName() {
        return groupName;
    }

    public void setGroupName(String groupName) {
        this.groupName = groupName;
    }

    public int getGroupType() {
        return groupType;
    }

    public void setGroupType(int groupType) {
        this.groupType = groupType;
    }

    public int getGroupOwner() {
        return groupOwner;
    }

    public void setGroupOwner(int groupOwner) {
        this.groupOwner = groupOwner;
    }

    public Groups() {
    }
}

File: src/main/kotlin/com/bandtec/finfamily/finfamily/model/Groups.kt
package com.bandtec.finfamily.finfamily.model

import javax.persistence.*
import io.swagger.annotations.*

@Entity
@Table(name = "groups")
class Groups(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @ApiModelProperty(required = false, hidden = true)
        val id : Int = 0,
        @ApiModelProperty(required = true, hidden = false)
        @Column(name = "group_name")
        val groupName : String = "",
        @ApiModelProperty(required = true, hidden = false, allowableValues = "1,2")
        @Column(name = "group_type")
        val groupType : Int = 0,
        @ApiModelProperty(required = true, hidden = false)
        @Column(name = "group_owner")
        val groupOwner : Int = 0,
        @Column(name = "group_external_id")
        @ApiModelProperty(required = false, hidden = true)
        var externalGroupId : String =  ""

)